File: service-web/src/main/java/com/yanggy/cloud/config/mvc/MvcConfig.java
package com.yanggy.cloud.config.mvc;

import com.alibaba.fastjson.serializer.SerializerFeature;
import com.alibaba.fastjson.support.config.FastJsonConfig;
import com.alibaba.fastjson.support.spring.FastJsonHttpMessageConverter;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.MediaType;
import org.springframework.http.converter.HttpMessageConverter;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurationSupport;

import java.util.ArrayList;
import java.util.List;

/**
 * @author derrick.yang
 * @Date 10/17/18 16:34
 */

@Configuration
public class MvcConfig extends WebMvcConfigurationSupport {
    @Override
    protected void configureMessageConverters(List<HttpMessageConverter<?>> converters) {
        super.configureMessageConverters(converters);

        FastJsonHttpMessageConverter fastConverter = new FastJsonHttpMessageConverter();

        // 1.添加fastJson的配置信息，是否要格式化返回的Json数据；
        FastJsonConfig fastJsonConfig = new FastJsonConfig();
        fastJsonConfig.setSerializerFeatures(SerializerFeature.PrettyFormat);

        // 2、处理中文乱码
        List<MediaType> fastMediaTypes = new ArrayList<>();
        fastMediaTypes.add(MediaType.APPLICATION_JSON_UTF8);
        fastConverter.setSupportedMediaTypes(fastMediaTypes);

        // 3、在convert中添加配置信息；
        fastConverter.setFastJsonConfig(fastJsonConfig);

        // 4、将convert添加到converters当中；
        converters.add(fastConverter);
    }
}

File: kothlin-service/src/main/kotlin/com/yanggy/cloud/kothlin/config/MvcConfig.kt
package com.yanggy.cloud.kothlin.config;

import com.alibaba.fastjson.serializer.SerializerFeature
import com.alibaba.fastjson.support.config.FastJsonConfig
import com.alibaba.fastjson.support.spring.FastJsonHttpMessageConverter
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType
import org.springframework.http.converter.HttpMessageConverter
import org.springframework.web.servlet.config.annotation.WebMvcConfigurationSupport

/**
 *@author derrick.yang
 *@Date 24/10/2018 16:49
 */

@Configuration
class MvcConfig : WebMvcConfigurationSupport() {
    override fun configureMessageConverters(converters: MutableList<HttpMessageConverter<*>>) {
        super.configureMessageConverters(converters)
        var fastConverter: FastJsonHttpMessageConverter = FastJsonHttpMessageConverter()

        var fastJsonConfig: FastJsonConfig = FastJsonConfig()
        fastJsonConfig.setSerializerFeatures(SerializerFeature.PrettyFormat)

        var fastMediaType: MutableList<MediaType> = mutableListOf()
        fastMediaType.add(MediaType.APPLICATION_JSON_UTF8)

        fastConverter.fastJsonConfig = fastJsonConfig
        fastConverter.supportedMediaTypes = fastMediaType

        converters.add(fastConverter)
    }
}